app/src/main/java/sdumchykov/androidApp/presentation/viewPager/contacts/MyContactsFragment.kt

```diff
@@ -0,0 +1,302 @@
+        name?.text?.clear()
+        surname?.text?.clear()
+        profession?.text?.clear()
+    }
+
+    private fun handleRecyclerViewContent() {
+        if (parentViewModel.getFetchContactList()) getContactsListWithDexter()
+    }
+
+    private fun getContactsListWithDexter() {
+        Dexter.withActivity(activity).withPermission(Manifest.permission.READ_CONTACTS)
+            .withListener(object : PermissionListener {
+                override fun onPermissionGranted(response: PermissionGrantedResponse) {
+                    if (response.permissionName == Manifest.permission.READ_CONTACTS) {
+                        fetchContacts()
+                    }
+                }
+
+                override fun onPermissionDenied(response: PermissionDeniedResponse) {
+                    Toast.makeText(
+                        activity, "Permission should be granted!", Toast.LENGTH_SHORT
+                    ).show()
+                }
+
+                override fun onPermissionRationaleShouldBeShown(
+                    permission: PermissionRequest,
+                    token: PermissionToken
+                ) {
+                    token.continuePermissionRequest()
+                }
+            }).check()
+    }
+
+    @SuppressLint("Range")
+    private fun fetchContacts() {
+        val phones = activity?.contentResolver?.query(
+            ContactsContract.CommonDataKinds.Phone.CONTENT_URI, null, null, null, null
+        )
+
+        if (phones != null) {
+            val userModels = ArrayList<UserModel>()
+            while (phones.moveToNext()) {
+                val name =
+                    phones.getString(phones.getColumnIndex(ContactsContract.CommonDataKinds.Phone.DISPLAY_NAME))
+                val phoneNumber =
+                    phones.getString(phones.getColumnIndex(ContactsContract.CommonDataKinds.Phone.NUMBER))
+                val contact = UserModel(userModels.size, name, phoneNumber, HARDCODED_IMAGE_URL)
+
+                userModels.add(contact)
+            }
+            parentViewModel.addData(userModels)
+            phones.close()
+        }
+    }
+
+    //endregion
+    private fun enableMultiselectViewState() {
+        with(binding) {
+            frameLayoutButtonsContainer?.visibility = View.VISIBLE
+            buttonRemoveSelectedContacts?.visibility = View.VISIBLE
+            textViewMyContactsAddContacts.visibility = View.GONE
+            imageViewMyContactsArrowBack.visibility = View.GONE
+            imageViewMyContactsCancel?.visibility = View.VISIBLE
+        }
+    }
+
+    private fun disableMultiselectViewState() {
+        with(binding) {
+            frameLayoutButtonsContainer?.visibility = View.GONE
+            buttonRemoveSelectedContacts?.visibility = View.GONE
+            textViewMyContactsAddContacts.visibility = View.VISIBLE
+            imageViewMyContactsArrowBack.visibility = View.VISIBLE
+            imageViewMyContactsCancel?.visibility = View.GONE
+        }
+    }
```

Comment:
create two extension functions: View.gone and View.visible and simplify the code